src/test/java/reactor/kafka/sender/internals/MockTransactionTest.java

```diff
@@ -133,10 +133,10 @@ public void transactionalReceiveAndSend() throws Exception {
         disposable.dispose();
         verifyTransaction(count, count);
 
-        assertEquals(transactionCount, producer.beginCount);
-        assertEquals(transactionCount, producer.commitCount);
-        assertEquals(0, producer.abortCount);
-        assertEquals(transactionCount, producer.sendOffsetsCount);
+        assertThat(producer.beginCount).as("beginCount").isGreaterThanOrEqualTo(transactionCount);
+        assertThat(producer.commitCount).as("commitCount").isGreaterThanOrEqualTo(transactionCount);
+        assertThat(producer.abortCount).as("abortCount").isZero();
+        assertThat(producer.sendOffsetsCount).as("sendOffsetsCount").isEqualTo(transactionCount);
     }
```

Comment:
@bsideup This should fix the flakiness - yes, it's possible you will get less then 10 per poll.

Unfortunately, there is no `poll.min.records`, you can only approximate it with `fetch.min.bytes`.

Reply:
Thanks! This is the assumption I had but wasn't sure about, appreciate the clarification 👍